torch/csrc/utils/tensor_new.cpp

```diff
@@ -212,7 +212,7 @@ Tensor internal_new_from_data(
     throw TypeError("new(): invalid data type '%s'", Py_TYPE(data)->tp_name);
   }
 
-  if (THPVariable_Check(data)) {
+  if (PyObject_IsInstance(data, THPVariableClass) == 1) {
```

Comment:
What's going on here? This wasn't here on my initial review

Reply:
Some test constructions were not considered in the first commit.  I am talking about  `sparse_csr_tensor` construction from lists. 
https://github.com/pytorch/pytorch/blob/b07d38a05df0ef7cf313b7751e86bce97692778d/test/test_sparse_csr.py#L36
Something weird was happening when I was trying to create these tensors, the function    `THPVariable_Check/PyObject_IsInstance` returned `-1` (an error) or `0` (the expected one). This caused segmentation faults, I tracked every function call from the stack and there is no changes in the input. That's why I changed this condition to a more restricted one, I could find any other solution.

Reply:
Hmm, I don't really understand this explanation. Why did returning -1/0 cause segmentation faults? Is it possible to print out what `data` is in the stack traces that caused the segfault?

Given that the line of code you quoted is `sparse_csr_tensor`, what seems more likely to me is that there is insufficient checking in the `sparse_csr_tensor` binding for, e.g., lists. If that's the case, you should fix it there, not here... (unless you can give a good argument for why this line of code is wrong)

Reply:
I figured out the error, the way how processing input parameters at `sparse_csr_tensor_ctor` functions were wrong. This change was reverted.